### PR TITLE
feat: define weasel app/entrypoints

### PR DESCRIPTION
**Pull request by bdura**
## Description

This PR creates a new entrypoint `weasel`, and enables `python -m weasel`.

It also ports `spacy project`-related tests to Weasel, and adapts them to the new tool.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -23,4 +23,11 @@ classifiers =
...
-    Source = https://github.com/explosion/weasel/
\ No newline at end of file
+    Source = https://github.com/explosion/weasel/
+
+[options.entry_points]
+console_scripts =
+    weasel = weasel._util:app
+    # TODO: Using 'spacy' as an alias has consequences,
+    # and may not be the best approache
+    spacy = weasel._util:app
```
Comment (svlandeg) [inline setup.cfg:33]: Is there a way to reverse the logic and have spaCy deal with this instead? Ideally, I'd want `weasel` to be ignorant of spaCy, if you know what I mean.

Comment (bdura) [reply setup.cfg:33]: Absolutely, it makes sense. As discussed IRL, we can do something like:

```python
from weasel.cli import app as project_cli

app.add_typer(project_cli, name="project")
```

on the spaCy side. That way `python -m spacy project` will seamlessly run Weasel under the hood.

I'll change this PR accordingly.

**Review by svlandeg — COMMENTED** [review]
What would be nice to have here is an additional "app" unit test like the ones we have in spaCy: https://github.com/explosion/spaCy/blob/master/spacy/tests/test_cli_app.py. These allow to test that the actual commands work. 

We could even just for now copy/paste some of these projects tests, so that we're literally testing that these work with `python -m spacy projects` as well as `python -m weasel`.

**Comment by bdura** [comment]
Thanks for the input, I will get on it!

**Comment by svlandeg** [comment]
One other suggestion I have is this: while we go through changes here, it might make sense to think about the related changes needed on the spaCy branch. If you could review https://github.com/explosion/spaCy/pull/12456 first, then after that we can have PRs to https://github.com/explosion/spaCy/tree/feature/weasel, I think it might be easier to do these in parallel bit by bit?

Because the `app` thing from this PR will be closely related to how we need to get `spacy projects` up and running again, right?

File: weasel/__main__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from ._util import COMMAND, app
+
+app(prog_name=COMMAND)
```
Comment (svlandeg) [inline weasel/__main__.py:3]: Is it convention to have this in `__main__`?

File: weasel/tests/test_cli_app.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    "assets": [
+        {
+            "dest": "assets/spacy-readme.md",
+            "url": "https://github.com/explosion/spaCy/raw/dec81508d28b47f09a06203c472b37f00db6c869/README.md",
+            "checksum": "411b2c89ccf34288fae8ed126bf652f7",
+        },
+        {
+            "dest": "assets/citation.cff",
+            "url": "https://github.com/explosion/spaCy/raw/master/CITATION.cff",
+            "checksum": "c996bfd80202d480eb2e592369714e5e",
```
Comment (svlandeg) [inline weasel/tests/test_cli_app.py:32]: Silly little thing, but let's have these tests refer to other files and not spaCy-repo-specific ones. Could be weasel's readme, for instance.

**Review by svlandeg — COMMENTED** [review]
Otherwise looks good! 🎉

File: weasel/__main__.py (hunk 2 above)
Comment (bdura) [reply weasel/__main__.py:3]: I'm not totally sure what you mean... Putting `app()` there is how `python -m <package>` operates: https://typer.tiangolo.com/tutorial/package/#add-a-__main__py

File: weasel/tests/test_cli_app.py (hunk 3 above)
Comment (bdura) [reply weasel/tests/test_cli_app.py:32]: Nice catch!

**Pull request merged**